cranelift/codegen/src/verifier/mod.rs

```diff
@@ -1972,6 +1977,50 @@ impl<'a> Verifier<'a> {
         if errors.has_error() { Err(()) } else { Ok(()) }
     }
 
+    fn verify_try_call_handler_index(
+        &self,
+        inst: Inst,
+        block: Block,
+        index_imm: i64,
+        errors: &mut VerifierErrors,
+    ) -> VerifierStepResult {
+        if index_imm < 0 {
+            return errors.fatal((
+                inst,
+                format!("exception handler index {index_imm} cannot be negative"),
+            ));
+        }
+        let Ok(index) = usize::try_from(index_imm) else {
+            return errors.fatal((
+                inst,
+                format!("exception handler index {index_imm} is out-of-range"),
+            ));
+        };
+        let Some(terminator) = self.func.layout.last_inst(block) else {
+            return errors.fatal((
+                inst,
+                format!("referenced block {block} does not have a terminator"),
+            ));
+        };
+        let Some(et) = self.func.dfg.insts[terminator].exception_table() else {
+            return errors.fatal((
+                inst,
+                format!("referenced block {block} does not end in a try_call"),
+            ));
+        };
+
+        let etd = &self.func.dfg.exception_tables[et];
+        let num_exceptional_edges = etd.all_branches().len() - 1;
+        if index >= num_exceptional_edges {
+            return errors.fatal((
+                inst,
+                format!("exception handler index {index_imm} is out-of-range"),
+            ));
+        }
```

Comment:
Is the -1 here to skip the non-exceptional block at the front? If so mind leaving a comment to that effect? (and maybe model this as `if etd.all_branches[1..].get(index).is_none()`)

Reply:
The non-exceptional block is actually stored last (I originally had it the other way but that came out of code-review discussion IIRC); but the `- 1` is just because for N out-edges there are N-1 handlers we can select. Will add a comment.